app/screens/permalink/permalink.tsx

```diff
@@ -269,19 +269,21 @@ function Permalink({
 
     const handlePress = useCallback(preventDoubleTap(() => {
         if (channel) {
-            switchToChannelById(serverUrl, channel?.id, channel?.teamId);
+            switchToChannelById(serverUrl, channel.id, channel.teamId);
         }
-    }), []);
+    }), [channel]);
 
     const handleJoin = useCallback(preventDoubleTap(async () => {
+        setLoading(true);
+        setError(undefined);
```

Comment:
I'm not sure I'm following correctly.  

If the state variable `error` is not `undefined`, then the `PermalinkError` will be shown.  In it, onPress will call the `handleJoin` function.  But in there, we are setting `setError(undefined)` but on the next line, we are checking if `error` has properties teamId/channelId....  It might be async but will it not be false?

Also, we are not setting the Loading to false at the end of the function.

Reply:
The thing is that `error` has the value that it had in the render were `handleJoin` was created, so even if we call the state change, this callback still has the value from when it was called.

Since we want to show an alert, and get back to "the previous state", we reset the error to what it was.

Reply:
Ok